includes/reader-activation/class-reader-activation.php

```diff
@@ -181,17 +172,79 @@ public static function enqueue_scripts() {
+			'invalid_email'           => __( 'Please enter a valid email address.', 'newspack-plugin' ),
+			'invalid_password'        => __( 'Please enter a password.', 'newspack-plugin' ),
+			'invalid_display'         => __( 'Display name cannot match your email address. Please choose a different display name.', 'newspack-plugin' ),
+			'blocked_popup'           => __( 'The popup has been blocked. Allow popups for the site and try again.', 'newspack-plugin' ),
+			'code_resent'             => __( 'Code resent! Check your inbox.', 'newspack-plugin' ),
+			'create_account'          => __( 'Create an account', 'newspack-plugin' ),
+			'signin'                  => [
+				'title'           => __( 'Sign in', 'newspack-plugin' ),
+				'success_title'   => __( 'Success! You’re signed in.', 'newspack-plugin' ),
+				'success_message' => __( 'Login successful!', 'newspack-plugin' ),
+				'continue'        => __( 'Continue', 'newspack-plugin' ),
+				'resend_code'     => __( 'Resend code', 'newspack-plugin' ),
+				'otp'             => __( 'Email me a one-time code instead', 'newspack-plugin' ),
+				'otp_title'       => __( 'Enter the code sent to your email.', 'newspack-plugin' ),
+				'forgot_password' => __( 'Forgot password', 'newspack-plugin' ),
+				'create_account'  => __( 'Create an account', 'newspack-plugin' ),
+				'register'        => __( 'Sign in to an existing account', 'newspack-plugin' ),
+				'go_back'         => __( 'Go back', 'newspack-plugin' ),
+				'set_password'    => __( 'Set a password (optional)', 'newspack-plugin' ),
+			],
+			'register'                => [
+				'title'               => __( 'Create an account', 'newspack-plugin' ),
+				'success_title'       => __( 'Success! Your account was created and you’re signed in.', 'newspack-plugin' ),
+				'success_description' => __( 'In the future, you’ll sign in with a magic link, or a code sent to your email. If you’d rather use a password, you can set one below.', 'newspack-plugin' ),
+			],
+			'verify'                  => __( 'Thank you for verifying your account!', 'newspack-plugin' ),
+			'magic_link'              => __( 'Please check your inbox for an authentication link.', 'newspack-plugin' ),
+			'password_reset_interval' => __( 'Please wait a moment before requesting another password reset email.', 'newspack-plugin' ),
+			'account_link'            => [
+				'signedin'  => __( 'My Account', 'newspack-plugin' ),
+				'signedout' => __( 'Sign In', 'newspack-plugin' ),
+			],
+			'newsletters'             => __( 'Subscribe to our newsletter', 'newspack-plugin' ),
+		];
+
+		if ( empty( self::$reader_auth_labels ) ) {
+			/**
+			* Filters the global labels for reader activation auth flow.
+			*
+			* @param mixed[] $labels Labels keyed by name.
+			*/
+			self::$reader_auth_labels = apply_filters( 'newspack_reader_activation_auth_labels', $default_labels );
+		}
+
+		if ( ! $key ) {
+			return array_merge( $default_labels, self::$reader_auth_labels );
```

Comment:
Kind of a nit: this should be a deep merge, if the filter is to be used as suggested in the PR description. With the following use, the whole `signin` key is replaced, and sub-keys' values other than `title` are empty:

```
		add_filter(
			'newspack_reader_activation_auth_labels',
			function( $labels ) {
				return [
					'signin' => [
						'title' => 'banana',
					],
				];
			}
		);
```

The proper use of the filter in this case should be:

```
		add_filter(
			'newspack_reader_activation_auth_labels',
			function( $labels ) {
				$labels['signin']['title'] = 'banana';
				return $labels;
			}
		);
``` 

Just pointing this out since there's an `array_merge` in use, suggesting _all_ defaults will be merged with what's supplied via the filter.

Reply:
Good suggestion!

Rather than perform a deep merge on every call to `get_reader_auth_labels`, I decided to adjust the logic for setting the labels cache to account for both usages of the filters above. Let me know what you think:

https://github.com/Automattic/newspack-plugin/pull/3070/commits/5b429756e0ca98f1daf1c0ddaff0783f982ac67c